### ISSUE TITLE
canPerformAction doesn't enable menu item

### ISSUE THREAD

**Issue by ghazel** [jessesquires/JSQMessagesViewController#626]
I'm trying to add a custom UIMenuItem for a cell. Just like #569

I'm adding the menu item via `setMenuItems` and returning `YES` from `canPerformAction`, but the item still does not appear in the menu.

It seems to work if I add the expected custom selector to `JSQMessagesCollectionViewCell`.

My code:

``` objc
- (void)viewDidLoad
{
    ...
    UIMenuController *menuController = [UIMenuController sharedMenuController];
    UIMenuItem *hooray = [[UIMenuItem alloc] initWithTitle:@"hooray"
                                                    action:@selector(hooray:)];
    [menuController setMenuItems:@[hooray]];
    ...
}

- (BOOL)collectionView:(UICollectionView *)collectionView canPerformAction:(SEL)action forItemAtIndexPath:(NSIndexPath *)indexPath withSender:(id)sender
{
    if (action == @selector(hooray:))) {
        return YES;
    }

    return [super collectionView:collectionView canPerformAction:action forItemAtIndexPath:indexPath withSender:sender];
}

- (void)collectionView:(JSQMessagesCollectionView *)collectionView performAction:(SEL)action forItemAtIndexPath:(NSIndexPath *)indexPath withSender:(id)sender
{
    if (action == @selector(hooray:)) {
        [self hooray:indexPath];
    } else {
        [super collectionView:collectionView performAction:action forItemAtIndexPath:indexPath withSender:sender];
    }
}
```

**Comment by buithuyen** [comment]
What exactly did you do in this case?
I have added and showed up custom item menu successfully. But when I tap this item, there is nothing happen.

I realize that `UICollectionView Delegate` didn't call 
`collectionView:performAction:forItemAtIndexPath:withSender:` method.

What should I do in this case? Thanks!

**Comment by eliburke** [comment]
@buithuyen there are some issues with custom menu items and behaviors that aren't slated to be fixed until the 8.0 branch is released. Check out PR #1563 -- perhaps you can incorporate that into your code for now.

**Comment by patniemeyer** [comment]
Is it still the case that there is no clean way to get the cell / index path of the item selected from collection view?  I also see that collectionView:performAction is never called.

As a simple workaround I am storing the indexpath of the item touched to display the menu when collectionView:canPeformAction is invoked.  (It actually seems to be invoked twice: once before showing the menu as I'd expect and once after the user selects the menu item.)
